src/fomo.jl

```diff
@@ -1,11 +1,121 @@
+const FOMO_η2 = 95 // 100
+const FOMO_γ1 = 1 // 2
+const FOMO_γ2 = 2
+const FOMO_γ3 = 1 // 2
+const FOMO_αmax = "1 / eps(T)" # TODO
+const FOMO_β = 9 // 10
+const FOMO_θ1 = 1 // 10
+const FOMO_θ2 = "eps(T)^(1 // 3)" # TODO
+const FOMO_M = 1
+const FOMO_step_backend = :r2_step
+
+"""
+    FOMOParameterSet{T} <: AbstractParameterSet
+
+This structure designed for `lbfgs` regroups the following parameters:
+  - `η1 = $(FOMO_η1)`, `η2 = T($(FOMO_η2))`: step acceptance parameters.
+  - `γ1 = T($(FOMO_γ1))`, `γ2 = T($(FOMO_γ2))`: regularization update parameters.
+  - `γ3 = T($(FOMO_γ3))` : momentum factor βmax update parameter in case of unsuccessful iteration.
+  - `αmax = $(FOMO_αmax)`: maximum step parameter for fomo algorithm.
+  - `β = T($(FOMO_β)) ∈ [0,1)`: target decay rate for the momentum.
+  - `θ1 = T($(FOMO_θ1))`: momentum contribution parameter for convergence condition (1).
+  - `θ2 = $(FOMO_θ2)`: momentum contribution parameter for convergence condition (2). 
+  - `M = $(FOMO_M)` : requires objective decrease over the `M` last iterates (nonmonotone context). `M=1` implies monotone behaviour. 
+  - `step_backend = $(FOMO_step_backend)()`: step computation mode. Options are `r2_step()` for quadratic regulation step and `tr_step()` for first-order trust-region.
+
+  Default values are:
+  - `η1::T = $(FOMO_η1)`
+  - `η2::T = T($(FOMO_η2))`
+  - `γ1::T = T($(FOMO_γ1))`
+  - `γ2::T = T($(FOMO_γ2))`
+  - `γ3::T = T($(FOMO_γ3))`
+  - `αmax::T = $(FOMO_αmax)`
+  - `β = T($(FOMO_β)) ∈ [0,1)`
+  - `θ1 = T($(FOMO_θ1))`
+  - `θ2 = e$(FOMO_θ2)`
+  - `M = $(FOMO_M)`
+  - `step_backend = $FOMO_step_backend()
+"""
+struct FOMOParameterSet{T} <: AbstractParameterSet
+  η1::Parameter{T, RealInterval{T}}
+  η2::Parameter{T, RealInterval{T}}
+  γ1::Parameter{T, RealInterval{T}}
+  γ2::Parameter{T, RealInterval{T}}
+  γ3::Parameter{T, RealInterval{T}}
+  αmax::Parameter{T, RealInterval{T}}
+  β::Parameter{T, RealInterval{T}}
+  θ1::Parameter{T, RealInterval{T}}
+  θ2::Parameter{T, RealInterval{T}}
+  M::Parameter{Int, IntegerRange{Int}}
+  step_backend::Parameter{Union{r2_step, tr_step}, CategoricalSet{Union{r2_step, tr_step}}}
+end
+
+# add a default constructor
+function FOMOParameterSet{T}(;
+  η1::T = eps(T)^(1 // 4),
+  η2::T = T(FOMO_η2),
+  γ1::T = T(FOMO_γ1),
+  γ2::T = T(FOMO_γ2),
+  γ3::T = T(FOMO_γ3),
+  αmax::T = 1 / eps(T),
+  β::T = T(FOMO_β),
+  θ1::T = T(FOMO_θ1),
+  θ2::T = eps(T)^(1 // 3),
+  M::Int = FOMO_M,
+  step_backend::AbstractFOMethod = eval(FOMO_step_backend)(),
+) where {T}
+  @assert η1 <= η2
+  FOMOParameterSet(
+    Parameter(η1, RealInterval(T(0), T(1), lower_open = true, upper_open = true)),
+    Parameter(η2, RealInterval(T(0), T(1), lower_open = true, upper_open = true)),
+    Parameter(γ1, RealInterval(T(0), T(1), lower_open = true, upper_open = true)),
+    Parameter(γ2, RealInterval(T(1), T(Inf), lower_open = true, upper_open = true)),
+    Parameter(γ3, RealInterval(T(0), T(1), lower_open = true, upper_open = true)),
```

Comment:
```suggestion
    Parameter(γ3, RealInterval(T(0), T(1))),
```
This one can be equal to 0 or 1.